### DISCUSSION TITLE
URL Link Preview?

### DISCUSSION THREAD

**Discussion by caschy** [TriliumNext/Trilium#3169] [Q&A] [answered]
Is there anyway to show a site preview when you post a url/link in a note?  I know there is a free api (http://www.linkpreview.net/) that returns page, title, desc, and an img for a given url.  Is there a way to integrate this into the note editor?  Thanks.

**Comment by zadam** [comment]
Hi, there isn't such a feature in Trilium.

**Comment by caschy** [comment]
Well ok, thanks for the response.

I did manage to get something working but its only a half solution really.

I've used a rendernote and an html code note to show the results of querying linkpreview.net.  It works fine but I'd like to know how to be able to paste this into other notes at will.  Here is the code for the html portion:

' 
        function httpGet(theUrl) {
              let xmlHttpReq = new XMLHttpRequest();
             theUrl = "https://api.linkpreview.net?key=xxxxxxxxxxxxxxxxxxxxxxxx&q="+theUrl
              xmlHttpReq.open("GET", theUrl, false); 
              xmlHttpReq.send(null);
          return xmlHttpReq.responseText;
         }
 
        function foo() {
            urlly= document.getElementById("url").value;
            response = httpGet(urlly);
            data =JSON.parse(response); 
            document.getElementById("tit").innerHTML = data.title;
           $("#pic1").attr("src", data.image);
            document.getElementById("desc").innerHTML = data.description;
            $("#linky").attr("href", urlly);
             return true;
          }
  '

**Comment by caschy** [comment, marked as answer]
Well just in case anyone else is looking for a solution to this I figured it out...sort of:

I'm using a combination of a Render note and an HTML code note.  In the HTML code note I have the following (see attached txt file)

The HTML code note renders a simple input box, submit button and div to show results.  Using the free linkpreview.net api, you can query a url to obtain a website title, website representative img and short description.  These are displayed in the div, along with the url, after you input a url in the inputbox and click the submit button.  I then use my browsers "web selection" to select the resultant div and copy it and then can paste it back into Trillium into whatever (new note, existing etc.).  I guess you could use the trillium webclipper extension in chrome to also do this.

Here is an example of what it looks like:

![image](https://user-images.githubusercontent.com/18508692/194562763-b797cd4f-8591-412c-872d-493f5eea6dc4.png)

[html code note.txt](https://github.com/zadam/trilium/files/9734230/html.code.note.txt)
